fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractProgressiveLoanScheduleGenerator.java

```diff
@@ -181,6 +181,13 @@ public LoanScheduleModel generate(final MathContext mc, final LoanApplicationTer
             // }
         }
 
+        // Fixed Length validation
+        final int lastPeriodIdx = periods.size() - 1;
+        LoanScheduleModelRepaymentPeriod lastPeriod = (LoanScheduleModelRepaymentPeriod) periods.get(lastPeriodIdx);
+        if (lastPeriod.getDueDate().compareTo(loanEndDate) != 0) {
```

Comment:
With reschedule, i dont think it will work... :/

Based on the periodNumber of scheduleParams, you can decide in the "nextRepaymentDate()" method of DefaultScheduledDateGenerator, whether it is the last installment or not and you can set the proper repayment date. After that if the installment was rescheduled, that date would be overriden, but i guess that is fine.

What do you think?

Reply:
Done! Updated to validate the last installment